splice_machine/src/main/java/com/splicemachine/derby/vti/ORCRecordIterator.java

```diff
@@ -0,0 +1,246 @@
+	}
+
+	/*
+	 * This checks if there are more records to process. First checks if there
+	 * are more records in current file. If no more records, gets the next file
+	 * and checks that for more records.
+	 */
+	@Override
+	public boolean hasNext() {
+		try {
+
+			// If no more records in current file
+			if (!records.hasNext()) {
+				// check if there is next file
+				if (this.hasNextFile()) {
+					// Next file exists, so get the records from that file
+					getNextFileRecords();
+				}
+			}
+
+			return records.hasNext();
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	/*
+	 * This formats the record in the ORC file and return as LocatedRow
+	 */
+	@Override
+	public LocatedRow next() {
+		try {
+			// Return Row Format
+			ExecRow returnRow = execRow.getClone();
+			// Get the row
+			row = records.next(row);
+
+			// Get the row details
+			List value_lst = inspector.getStructFieldsDataAsList(row);
+
+			String value = null;
+			// Construct row for each column
+			for (int i = 1; i <= execRow.nColumns(); i++) {
+				Object field = value_lst.get(i - 1);
+				value = null;
+				if (field != null)
+					value = field.toString();
+				// Note: Date is in same format as Splice Default,
+				// For timestamp format yyyy-mm-dd hh:mm:ss[.f...]., Check if
+				// Timestamp.valueOf(value) is required
+				returnRow.getColumn(i).setValue(value);
+			}
+			return new LocatedRow(returnRow);
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	/*
+	 * This checks if there are more files to process in the folder
+	 */
+	public boolean hasNextFile() {
+		try {
+			// If it is folder, check the file list iterator for more files
+			if (isDir)
+				return fileList.hasNext();
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+		// If it is single file, return false
+		return false;
+	}
+
+	/*
+	 * Get the the next ORC fiel to process in the folder
+	 */
+	public Path nextFile() {
+		try {
+			// IF it is folder, get the next file from the file list iterator
+			if (isDir)
+				return fileList.next().getPath();
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+		return null;
+	}
+
+	/*
+	 * Returns Records in the next file
+	 */
+	public void getNextFileRecords() throws IOException {
+		// Close the current records iterator to release the resources for
+		// current file
+		records.close();
+
+		// Get the Record Reader for the next file
+		Reader reader = getReader(this.nextFile());
+		// set the records iterator for the new file
+		this.records = reader.rows();
+	}
+
+	/*
+	 * Gets the RecordReader of the specified file
+	 */
+	private Reader getReader(Path toProcessPath) throws IOException {
+		Reader reader = null;
+		if (toProcessPath == null) {
+			throw new IOException("Null File Path");
+		}
+		reader = OrcFile.createReader(filesystem, toProcessPath);
+
+		return reader;
+
+	}
+
+	@Override
+	public void remove() {
+		throw new RuntimeException("Not Implemented");
```

Comment:
Please throw new UnsupportedOperationException() instead, since we won't be supporting removal.